pkg/action/executor/k8s_controller_updater.go

```diff
@@ -164,6 +166,15 @@ func (c *k8sControllerUpdater) update(desired *controllerSpec) error {
 	}
 	glog.V(2).Infof("Successfully updated %v %s/%s",
 		c.controller, c.namespace, c.name)
+	if utilfeature.DefaultFeatureGate.Enabled(features.AllowIncreaseNsQuota4Resizing) {
+		isReady, err := c.controller.waitForReplicaSetToBeScheduled(c.name)
+		if !isReady {
+			glog.V(2).Infof("Timeout on waiting for new replicasets to be scheduled for the workload controller %s/%s", c.namespace, c.name)
+			return err
+		}
+		glog.V(2).Infof("All of the new replicasets get scheduled for the workload controller %s/%s", c.namespace, c.name)
+	}
```

Comment:
I think this section should be moved to inside `parentController.update`, and I feel we should not call `waitForReplicaSetToBeScheduled` for gitOps actions.

Reply:
make sense. will do